### PR TITLE
 add tailoring_file module

### PR DESCRIPTION
**Pull request by evgeni**
Fixes: #749

### CODE REVIEW

**Comment by evgeni** [comment]
no tests, no docs, but that's what I meant by inheritance

cc @ogajduse

**Comment by evgeni** [comment]
I wonder if this should be `foreman_scap_tailoring_file` instead. Or maybe even both modules `foreman_openscap_…`? We've been using the resource name so far as a guideline, but outside of Foreman and Katello, things get wild.

**Comment by evgeni** [comment]
That said, this is ready for review.

@jameerpathan111 thoughts on this?

**Comment by jameerpathan111** [comment]
> I wonder if this should be `foreman_scap_tailoring_file` instead. Or maybe even both modules `foreman_openscap_…`? 

`foreman_scap_tailoring_file` seems good to me.

File: plugins/module_utils/foreman_helper.py [hunk 1]
```diff
@@ -1017,6 +1019,39 @@ def __init__(self, **kwargs):
...
+class ForemanDataStreamModule(ForemanTaxonomicEntityAnsibleModule):
+    def __init__(self, **kwargs):
+        foreman_spec = dict(
+            original_filename=dict(type='str'),
+            scap_file=dict(type='path'),
+        )
+        foreman_spec.update(kwargs.pop('foreman_spec', {}))
+        super(ForemanDataStreamModule, self).__init__(foreman_spec=foreman_spec, **kwargs)
+
+    def datastream_run(self):
```
Comment (mdellweg) [inline plugins/module_utils/foreman_helper.py]: Why don't you override the actual 'run' method?

Comment (evgeni) [reply plugins/module_utils/foreman_helper.py]: because I'm a lazy bum and hate Python's `super`.

File: plugins/module_utils/foreman_helper.py [hunk 2]
```diff
@@ -1028,7 +1028,7 @@ def __init__(self, **kwargs):
         foreman_spec.update(kwargs.pop('foreman_spec', {}))
         super(ForemanDataStreamModule, self).__init__(foreman_spec=foreman_spec, **kwargs)
 
-    def datastream_run(self):
+    def run(self):
```
Comment (mdellweg) [inline plugins/module_utils/foreman_helper.py]: I know lazy n stuff...

But should it not support the same interface as the old one?

Comment (evgeni) [reply plugins/module_utils/foreman_helper.py]: I don't see a benefit implementing this at this point, tbh :)

Comment (mdellweg) [reply plugins/module_utils/foreman_helper.py]: Good for you, that the 'new' run does not accept parameters anyway.

**Comment by evgeni** [comment]
I'll merge this *after* #747 and handle the needed rebasing.

**Comment by evgeni** [comment]
FTR, the rebased version of this ontop of #747 can be found in https://github.com/evgeni/foreman-ansible-modules/tree/tailoring_entity, will replace this PR with it as soon as 747 lands.

File: plugins/doc_fragments/foreman.py [hunk 3]
```diff
@@ -276,3 +276,19 @@ class ModuleDocFragment(object):
     required: true
     type: str
 '''
+
+    DATASTREAM = '''
```
Comment (mdellweg) [inline plugins/doc_fragments/foreman.py]: Should this be named `SCAP_DATASTREAM` as it is rather specific?

Comment (evgeni) [reply plugins/doc_fragments/foreman.py]: yes

Comment (evgeni) [reply plugins/doc_fragments/foreman.py]: I've also renamed the helper class to match

**Comment by evgeni** [comment]
also squashed the commits to make it nicer :)
merging on green ci

**Pull request merged**